scripts/navbar_menu.js

```diff
@@ -0,0 +1,33 @@
+/* exported toggleMenu */
+
+// Toggle a menu contents to visible and rotate the menu icon
+function toggleMenu(menuId) {
+    let menu = document.getElementById(menuId);
+    [menu.getElementsByClassName("menu-contents")].forEach(element => {
+        element[0].classList.toggle("invisible");
+        element[0].classList.toggle("transparent");
+    });
+    [menu.getElementsByClassName("menu-icon")].forEach(element => {
+        element[0].classList.toggle("rotate");
+    });
+}
+
+// Close all dropdown menus if the user clicks outside of one
+document.addEventListener("click", function (event) {
+    let inMenu = false;
+    document.querySelectorAll(".menu").forEach(element => {
+        if (element.contains(event.target) || element == event.target) {
+            inMenu = true;
+        }
+    });
+
+    if (!inMenu) {
+        document.querySelectorAll(".menu-contents").forEach(element => {
```

Comment:
This duplicates the actions of `toggleMenu`. Can we reword this as something like
```
document.querySelector(".menu").forEach(element => toggleMenu(element.id));
```

Reply:
Ah, but it doesn't actually. `toggleMenu()` just toggles. If we called `toggleMenu()` it would close any open menus and open any closed menus, it wouldn't close everything. If there were 1 and only 1 menu they would be the same, but with more than 1 we can't use it.

Reply:
Ah. Yeah, I didn't read closely enough. Good point!